crates/store/re_log_types/src/index/absolute_time_range.rs

```diff
@@ -213,106 +213,6 @@ impl AbsoluteTimeRangeF {
     pub fn to_int(self) -> AbsoluteTimeRange {
         AbsoluteTimeRange::new(self.min.floor(), self.max.ceil())
     }
-
-    /// In the given range, pick the "simplest" number.
-    ///
-    /// The simplest number is defined approximately as
-    /// "the number with the most trailing zeroes".
-    ///
-    /// This is used in GUI code.
-    ///
-    /// NOTE: this function only cares about simple _integers_.
-    /// We don't care about "smart" selection things smaller than a nanosecond or a single time step.
-    /// So if both ends are within the same integer, then [`Self::center`] is returned.
-    pub fn smart_aim(self) -> TimeReal {
```

Comment:
This was an over-complicated way of doing snapping, that in the end resulted in worse behavior. The new algo uses the fact that the time panel is (locally) linear.

The original smart aiming algo came from egui/emath, but I improved it when I re-implemented it here.
These improvements have now been upstreamed: https://github.com/emilk/egui/pull/7680

(it still makes sense to use the smart-aiming approach in egui for logarithmic sliders and other non-linear situations)

Reply:
nice!